provider/cmd/pulumi-resource-nuage/nuage_provider/container_function.py

```diff
@@ -136,14 +141,14 @@ def __init__(
         auth = aws.ecr.get_authorization_token(registry_id=registry_id)
 
         self.image_uri = pulumi.Output.all(
-            url=args.repository_url, name=self.name
+            url=args.repository_url, name=self.full_name
```

Comment:
Where is `full_name` being assigned ? I don't see it in the code

Reply:
I understand how your fix resolves the problem, but I'd rather we don't add a "technical" variable into the component's namespace

Could you perhaps not bind it to `self` ?
Also, in terms of naming for anybody reading our code, maybe `_name` is better here